cms/djangoapps/contentstore/views/component.py

```diff
@@ -137,6 +146,40 @@ def container_handler(request, usage_key_string):
             section = get_parent_xblock(subsection)
             assert section is not None, "Could not determine ancestor section from unit " + six.text_type(unit.location)
 
+            prev_url = next_url = None
+            prev_loc = next_loc = None
+            last_block = None
+            siblings = list(section.get_children())
+            for i, block in enumerate(siblings):
+                if block.location == subsection.location:
+                    if last_block:
+                        try:
+                            prev_loc = last_block.get_children()[0].location
+                        except IndexError:
+                            pass
+                    try:
+                        next_loc = siblings[i + 1].get_children()[0].location
+                    except IndexError:
+                        pass
+                    break
+                last_block = block
+            if not prev_loc:
+                sections = course.get_children()
+                prev_sec_idx = sections.index(section)
+                if prev_sec_idx:
+                    prev_section = sections[prev_sec_idx - 1]
+                    prev_loc = prev_section.get_children()[-1].get_children()[-1].location
+            if not next_loc:
+                sections = course.get_children()
+                try:
+                    next_section = sections[sections.index(section) + 1]
+                    next_loc = next_section.get_children()[0].get_children()[0].location
+                except IndexError:
+                    pass
```

Comment:
I think this whole piece would benefit from either being in a helper method or having some comments to indicate what it's doing.

Reply:
I moved it to `contentstore.utils.get_sibling_urls`